Kudu.Tests/Core/Function/KafkaTriggerKedaAuthProviderTest.cs

```diff
@@ -9,16 +9,33 @@ namespace Kudu.Tests.Core.Function
 {
     public class KafkaTriggerKedaAuthProviderTest
     {
+        [Fact]
+        public void TestPopulateAuthenticationRef_Is_BindingCaseInsensitive()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""Protocol"": ""SaslSsl"",
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Equal(1, authRef.Count);
```

Comment:
Don't we need to validate the authRef it's self? it looks count only. 
If you want to write the pattern simply, you might use https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/

Reply:
authRef is simply going to be a key value pair of  "name": "<functionappname>" , So went with count. Testing the content seems to be a better idea. Modified to validating the authRef object data.